runtime/fastly/builtins/body.cpp

```diff
@@ -60,15 +60,15 @@ bool FastlyBody::read(JSContext *cx, unsigned argc, JS::Value *vp) {
     return false;
   }
   uint32_t chunkSize = std::round(chunkSize_val);
-  if (chunkSize < 0) {
-    JS_ReportErrorUTF8(cx,
-                       "FastlyBody.read: The `chunkSize` argument has to be a positive integer.");
-    return false;
-  }
 
   auto body = host_body(self);
   auto result = body.read(chunkSize);
   if (auto *err = result.to_err()) {
+    if (chunkSize < 0) {
```

Comment:
I don't think it's possible for this check to ever fail since `chunkSize` is a `uint32_t` which can't be negative. It seems like we should remove it or fix the validation? Is this validation path tested?

Reply:
Thanks for this, it showed that we don't have tests for this class. I've gone ahead and written some tests and fixed bugs found by the tests 👍